karma.conf.js

```diff
@@ -1,13 +1,120 @@
-const base = require('skatejs-build/karma.conf');
-const webpackConfig = require('./webpack.config');
-module.exports = (config) => {
-  base(config);
-
-  // Ensure mobile browsers have enough time to run.
-  config.browserNoActivityTimeout = 60000;
-
-  // Since we override the Webpack config, we must make sure Karma gets it.
-  config.webpack = Object.assign({}, config.webpack, webpackConfig, {
-    entry: undefined
-  });
+/**
+ * Overwritten karma.conf.js
+ * includes karma-typescript-preprocessor
+ */
+
+const cwd = process.cwd();
+const path = require('path');
+const browsers = require(path.join(cwd, 'karma.browsers'));
+const webpackConfig = require(path.join(cwd, 'webpack.config'));
+
+module.exports = function (config) {
+  // list of files / patterns to load in the browser
+  // all dependancies should be traced through here
+  let files = ['test/unit.js'];
+
+  // preprocess matching files before serving them to the browser
+  // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
+  // webpack will trace and watch all dependancies
+  let preprocessors = {
+    '**/*.ts': ['typescript'],
+    'test/unit.js': ['webpack', 'sourcemap']
+  };
+
+  if (process.argv.indexOf('--perf') > -1) {
+    files = ['test/perf.js'];
+    preprocessors = {
+      'test/perf.js': ['webpack', 'sourcemap']
+    };
+  }
+
+  config.set(Object.assign({
+    // base path that will be used to resolve all patterns (eg. files, exclude)
+    // setting to process.cwd will make all paths start in current component directory
+    basePath: process.cwd(),
+
+    // frameworks to use
+    // available frameworks: https://npmjs.org/browse/keyword/karma-adapter
+    frameworks: ['mocha', 'chai', 'sinon-chai'],
+
+    // list of files / patterns to load in the browser
+    files,
+
+    // list of files to exclude
+    exclude: [],
+
+    // list of preprocessors
+    preprocessors,
+
+    typescriptPreprocessor: {
```

Comment:
Can't we reuse the base config from skatejs-build still and just set `webpack` to `null` and set teh `typescriptPreprocessor to this?` I'm probably missing a couple things, but it would seem we can reuse quite a bit still.